datadog_lambda/handler.py

```diff
@@ -15,6 +18,27 @@ class HandlerError(Exception):
     pass
 
 
+class _ErrorOutsideHandlerDecorator(object):
+    """
+    Decorator for when an exception occurs outside of the handler function.
+    Emits telemetry and re-raises the exception.
+    """
+
+    def __init__(self, exception, modified_mod_name, start_time_ns):
+        self.exception = exception
+        self.modified_mod_name = modified_mod_name
+        self.start_time_ns = start_time_ns
+
+    def __call__(self, event, context, **kwargs):
+        emit_telemetry_on_exception_outside_of_handler(
+            context,
+            self.exception,
+            self.modified_mod_name,
+            self.start_time_ns,
+        )
+        raise self.exception
```

Comment:
This `_ErrorOutsideHandlerDecorator` is incredibly clever. Very well done.

I'd recommend we move this to `datadog_lambda.wrapper`. There's already a couple of wrapper classes in there. Then, rename it to something that similarly matches the casing etc of `datadog_lambda_wrapper`.